app/src/main/java/com/example/task/ToDoFragment.kt

```diff
@@ -16,26 +26,33 @@ class ToDoFragment : Fragment() {
 
     private val sharedViewModel: TaskViewModel by activityViewModels()
 
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var taskList: LiveData<List<Task>>
     private var _binding: FragmentTodoBinding? = null
-    // This property is only valid between onCreateView and
-    // onDestroyView.
     private val binding get() = _binding!!
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-
         _binding = FragmentTodoBinding.inflate(inflater, container, false)
         return binding.root
 
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
+                recyclerView = binding.recyclerViewForFragmentToDo
+                recyclerView.layoutManager = LinearLayoutManager(activity)
+                lifecycleScope.launch{
+                    TaskViewModel().getData().observe(viewLifecycleOwner,
+                        Observer {
```

Comment:
Don't need to write Observer. Just do this:

```kotlin
viewModel.someLiveData.observe(viewLifecycleOwner) {
    recyclerView.adapter = PendingAdapter(it)
```

This uses the kotlin SAM syntax.

Reply:
Okay, Sure